cmd/registry/controller/list.go

```diff
@@ -0,0 +1,134 @@
+package controller
+
+import (
+	"context"
+
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/cmd/registry/patterns"
+	"github.com/apigee/registry/pkg/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/registry/names"
+)
+
+type Lister interface {
```

Comment:
1. Why does Lister exist? (for testability?)
2. Why is Lister defined in the controller and not at some more general level (`core` or the top-level `pkg` directory)?

Reply:
The lister is an interface which can be faked while testing time sensitive controller features, specially the updated timestamp condition. It is currently defined in the controller because the interface includes only the functions which are used by the controller.

I explored the option of defining a generic registry client interface, but that complicates the  process where faking it means implementing almost all of the registry's functions. 

This is a much lighter implementation where I only need to care about the functions that the controller is using.

Reply:
Maybe `RegistryLister` belongs in `resolve.go` then. The controller defines the `Lister` interface to tell `resolve.go` and `controller-timestamp_test.go` what it needs to fetch resources. Then `resolve.go` and `controller-timestamp_test.go` have their own types that satisfy that interface.

As I said before though, if the types become useful in other places, we can move them to a more general place like `core`, `pkg`, etc. as Tim suggested.

Thoughts on this approach?

Reply:
I think what you are suggesting makes sense to me. Just to mention here, there is going to be a similar refactor for the scoring package where we will define an interface to make registry calls and update the timestamp condition. The `lister` interface and the one I define for scores might have a potential of overlap. 

So I am proposing that we keep things as is here for now. Once I define the interface for scoring package, we can do either of the following two things:
* merge the two interfaces and move the definition of RegistryLister (might be renamed) to a common package like `core`.
* Or if we decide that there is not enough overlap, move the `RegistryLister` definition to `resolve.go`.